src/main/java/com/diamondfire/helpbot/bot/command/impl/other/SamQuotesCommand.java

```diff
@@ -28,28 +34,184 @@ public HelpContext getHelpContext() {
                 .category(CommandCategory.OTHER);
     }
     
-    @Override
-    public ArgumentSet compileArguments() {
-        return new ArgumentSet();
-    }
-    
     @Override
     public Permission getPermission() {
         return Permission.USER;
     }
     
     @Override
     public void run(CommandEvent event) {
-        String[] strings = ExternalFiles.SAM_DIR.list();
-        File file = new File(ExternalFiles.SAM_DIR, strings[random.nextInt(strings.length)]);
-        EmbedBuilder builder = new EmbedBuilder();
-        builder.setTitle("Sam Quote");
-        builder.setImage("attachment://quote.png");
-        builder.setColor(new Color(87, 177, 71));
         
-        event.getChannel().sendMessage(builder.build()).addFile(file, "quote.png").queue();
+        if (event.getArgument("action") == "submit") {
+            
+            String[] message = event.getMessage().getContentRaw().split("/");
+            
+            long channelID = Long.parseLong(message[5]);
+            long messageID = Long.parseLong(message[6]);
+            
+            Objects.requireNonNull(event.getGuild().getTextChannelById(channelID)).retrieveMessageById(messageID).queue((messageText) -> {
```

Comment:
Objects.requireNonNull isn't needed. Ignore your IDE when it bugs you. Instead, tell the user if the channel they submitted is null.